content/_code-samples/markers-and-pagination/js/pagination-with-markers.js

```diff
@@ -0,0 +1,42 @@
+const xrpl = require("xrpl")
+
+async function main() {
+
+  // Create a client and connect to the network.
+  const client = new xrpl.Client("wss://xrplcluster.com/")
+  await client.connect()
+
+  // Specify a ledger to query for info.
+  let ledger = await client.request({
+    "command": "ledger_data",
+    "ledger_index": 500000,
```

Comment:
This is not quite the best practice that we want to showcase in this code sample.

There are three shortcut strings you can use when [specifying ledgers](https://xrpl.org/basic-data-types.html#specifying-ledgers), most notably `"validated"` meaning the latest validated ledger version. Typically when you want to query the ledger, you want to get the latest validated data, so this shortcut string is important.

However, it's possible that a new ledger version can get validated while you're iterating over multiple pages with a marker—the more pages, the more likely this is to happen. If you continue to specify `"validated"` your queries might start paging through the _new_ ledger instead, which might mean you get inconsistent data, or something is skipped or repeated because of the differences between the underlying ledgers. You might also get an error saying that your marker is invalid.

So, the best practice to avoid this problem is to use `"ledger_index": "validated"` on the first request (the one with no marker). Then, the response tells you what ledger version it used, so you can save the `ledger_index` or `ledger_hash` from that response (alongside the marker) and specify that same index or hash on each subsequent request.

Reply:
My mistake. I meant to do this, but for the sake of seeing the code run from start to finish I used a lower index. Updated code to use most recent validated ledger.